apps/web/scripts/kernel/handlers/kernel.ts

```diff
@@ -1,17 +1,23 @@
 import { type MessengerCallback, RequestType } from '@walless/messaging';
 
-import { handleConnect } from './connect';
+import { handleConnect, handleResolveConnect } from './connect';
 import * as solanaHandler from './solanaHandler';
 import * as suiHandler from './suiHandler';
 
 export const onKernelMessage: MessengerCallback = async (payload, channel) => {
 	if (payload.requestId) {
 		if (payload.type === RequestType.REQUEST_CONNECT) {
 			handleConnect(payload, channel);
+		} else if (payload.type === RequestType.RESOLVE_REQUEST_CONNECT) {
+			handleResolveConnect(payload, channel);
 		} else if (payload.type == RequestType.GET_ENDPOINT_ON_SOLANA) {
 			solanaHandler.getEndpoint(payload, channel);
 		} else if (payload.type === RequestType.SIGN_MESSAGE_ON_SOLANA) {
 			solanaHandler.handleSignMessage(payload, channel);
+		} else if (payload.type === RequestType.REQUEST_PAYLOAD) {
+			solanaHandler.handleRequestPayload(payload, channel);
+		} else if (payload.type === RequestType.RESOLVE_REQUEST_SIGNATURE) {
+			solanaHandler.handleResolveRequestSignature(payload, channel);
```

Comment:
Working with sign and send (both transaction and message), I think we need mechanism like `forwarding`. It means before passing to handler, we weed a forwarder to decide next action (like open popup). This mechanism will detach the handler at kernel with comunication between `sdk`, `kernel` and `ui`

Reply:
Go a head if you have concrete idea, will be useful to provide code snippet or discuss offline, I don't really understand what you're mentioning.